shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
Oh, I think I know why `close()` doesn't necessarily get called if the capability wrapper is constructed: The capability wrapper counts as a refcound and thus close() wouldn't be called until it is GC'd. Ouch. I think to fix this, node-capnp has to be adjusted to assume ownership of all returned capability wrappers (requiring the returner to use .clone() if they intended to keep a reference themselves).